gravityflow/gravityflow
# Retain non-editable pricing field inputs when needed for other pricing fields to function

## Description
re: [HS#11600](https://secure.helpscout.net/conversation/1010258681/11600?folderId=1113535)

Currently if you select a Total field or GP eCommerce Field (Subtotal, Tax, Discount) as editable they require the other pricing fields to be selected as editable as well for those fields to display the correct result.

`Gravity_Flow_Entry_Editor` has been updated so that if the form contains an editable Total, Coupon, Subtotal, Tax, or Discount field the inputs for all other pricing fields will be retained even if they are not selected as editable or display. Retained fields which are not selected as editable or display will be hidden by the gravityflow-hidden-field class.

## Testing instructions
Test as many form and user input step configurations as you can think of, covering the Product, Option, Quantity, Shipping, Total, Coupon, Subtotal, Tax, and Discount fields.

~~Note: the Coupon field does not currently restore the existing entry value, however, this will be resolved by https://github.com/gravityforms/gravityformscoupons/pull/13~~
Update: Populating the Coupon field with the existing entry value works with Coupons Add-On v2.9.2+. 

Please check that the coupon usage count displayed on the Forms > Coupons page is updated when adding or removing coupons on the User Input step.

## Automated Test Enhancements
Acceptance tests added:
- 0041PricingFieldsGFMergeTagCept: Tests that the GF front-end merge tag scripts can access the values
- 0041PricingFieldsTotalCept: Tests that the GF pricing scripts can access the values
- 0042CouponFieldCept: Tests that the Coupon field is functional
- 0042CouponFieldPopulatedCept: Tests that the entry value is populated for the Coupon field
- 0042CouponFieldUsageCountCept: Tests that the coupon usage counts are updated

## Documentation Changes?
https://docs.gravityflow.io/article/129-the-coupon-field-isnt-working will need updating to indicate the coupon field no longer requires any other display or editable fields.

## Checklist:
